openmc/data/photon.py

```diff
@@ -580,6 +643,89 @@ def from_endf(cls, photoatomic, relaxation=None):
 
         return data
 
+    @classmethod
+    def from_hdf5(cls, group_or_filename):
+        """Generate photon reaction from an HDF5 group
+
+        Parameters
+        ----------
+        group_or_filename : h5py.Group or str
+            HDF5 group containing interaction data. If given as a string, it is
+            assumed to be the filename for the HDF5 file, and the first group is
+            used to read from.
+
+        Returns
+        -------
+        openmc.data.IncidentPhoton
+            Photon interaction data
+
+        """
+        if isinstance(group_or_filename, h5py.Group):
+            group = group_or_filename
+        else:
+            h5file = h5py.File(str(group_or_filename), 'r')
+
+            # Make sure version matches
+            if 'version' in h5file.attrs:
+                major, minor = h5file.attrs['version']
+                # For now all versions of HDF5 data can be read
+            else:
+                raise IOError(
+                    'HDF5 data does not indicate a version. Your installation '
+                    'of the OpenMC Python API expects version {}.x data.'
+                    .format(HDF5_VERSION_MAJOR))
+
+            group = list(h5file.values())[0]
+
+        Z = group.attrs['Z']
+        data = cls(Z)
+
+        # Read energy grid
+        energy= group['energy'].value
+
+        # Read cross section data
+        for mt, (name, key) in _REACTION_NAME.items():
+            if key in group:
+                rgroup = group[key]
+            elif key in group['subshells']:
+                rgroup = group['subshells'][key]
+            else:
+                continue
+
+            data.reactions[mt] = PhotonReaction.from_hdf5(rgroup, mt, energy)
+
+        # Check for necessary reactions
+        for mt in [502, 504, 522]:
+            assert mt in data, "Reaction {} not found".format(mt)
+
+        # Read atomic relaxation
+        data.atomic_relaxation = AtomicRelaxation.from_hdf5(group['subshells'])
+
+        # Read Compton profiles
+        if 'compton_profiles' in group:
+            rgroup = group['compton_profiles']
+            profile = data.compton_profiles
+            profile['num_electrons'] = rgroup['num_electrons'].value
+            profile['binding_energy'] = rgroup['binding_energy'].value
```

Comment:
It seems the [`dataset[()]` is equivalent to `dataset.value`?](https://github.com/h5py/h5py/blob/master/h5py/_hl/dataset.py#L310-L314). The document says [x[()] returns a scalar if x is zero dimensional and a view otherwise. On the other hand x[...] always returns a view.](https://docs.scipy.org/doc/numpy/reference/arrays.indexing.html)

Reply:
I'm going to replace all `dataset.value` with `dataset[()]` if that is OK.

Reply:
Yes, I made the same realization after I made that comment! Thanks for pointing that out.